Add sample that summarizes Discord Forum Channel Threads (#69)
ballerina-guides/ai-samples: discord_thread_summarizer_using_openai/Readme.md

Reviewer: Should be README.md

Author: Renamed

# Summarize Forum Channel threads in Discord using OpenAI

## Use case

> Read and summarize Discord Forum Channel threads that has activity within the last 24 hours using OpenAI GPT 3.5-turbo model.

Reviewer: ```suggestion
Read and summarize Discord Forum Channel threads that have had activity within the last 24 hours using OpenAI GPT 3.5-turbo model.
```

or

```suggestion
Read and summarize Discord Forum Channel threads that have activity within the last 24 hours using OpenAI GPT 3.5-turbo model.
```

## Prerequisites
* OpenAI account
* Discord Bot with Read access to a Discord server
* Channel ID of the Discord Forum Channel

### Setting up OpenAI account
1. Create an [OpenAI](https://platform.openai.com/) account.
2. Go to **View API keys** and create a new secret key.

### Setting up the Discord Bot

> 1. Follow the steps mentioned in Step 1 of the [Discord Developer Portal](https://discord.com/developers/docs/getting-started#step-1-creating-an-app) to create a new App and a Discord Bot with `Read Messages/View` Channels.

Reviewer: Is this sentence complete? Is the last part correct/complete?

Author: Yes this is correct

Reviewer: Aren't these privileges?

Author: Fixed

2. Make sure you copy the Bot token and keep it safe.
3. In the same page, scroll down and enable `MESSAGE CONTENT INTENT` in the Bot settings so that the Bot can read the message content.
4. Install the app to your Discord server so that the Bot can read the messages.

> **Note:** The Steps 1, 2 and 4 in this Readme are all mentioned in the [Discord Developer Portal](https://discord.com/developers/docs/getting-started#step-1-creating-an-app).

Reviewer: ```suggestion
**Note:** The steps 1, 2, and 4 in this README are all mentioned in the [Discord Developer Portal](https://discord.com/developers/docs/getting-started#step-1-creating-an-app).
```

### Getting the Channel ID of the Discord Forum Channel
1. Follow the steps mentioned in [this article](https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-) to enable the Developer Mode in Discord.

> 2. Right click on the Discord Forum Channel and click on `Copy Channel ID` to copy the Channel ID.

Reviewer: Shall we stick to backticks or bold consistently for emphasis. https://github.com/ballerina-guides/ai-samples/blob/59aea9bda909054c8bca4433c79691e50b02e0cb/discord_thread_summarizer_using_openai/Readme.md#setting-up-openai-account, point 2 uses bold.

Author: Addressed

## Configuration

> Create a file called `Config.toml` at the root of the project.

Reviewer: ```suggestion
Create a file called `Config.toml` at the root of the project and add the relevant configuration.
```
